### DISCUSSION TITLE
Any idea why does `signInWithPassword` bypass the captcha token challenge

### DISCUSSION THREAD

**Discussion by razonyang** [supabase/supabase#30668] [Questions] [answered]
Hi, I turned on the captcha feature but it's not checking the captcha token. Is there a step I missed?

![image](https://github.com/user-attachments/assets/38bab28d-4227-4748-b97e-7073cc453776)

```js
    const { data, error } = await client.auth.signInWithPassword({
      email,
      password,
      options: {
        captchaToken: 'faketoken', 
      },
    })

    console.log(data, error) // the error is null.
```

**Comment by GaryAustin1** [comment]
You don't say if you followed all the steps.
https://supabase.com/docs/guides/auth/auth-captcha

**Reply by razonyang** [reply]
Thank you for your response. I followed the 1-2 steps on that guide:

1. `Sign up for Captcha` (Cloudflare Turnstile)
2. `Enable Captcha protection for your Supabase project`, as the image shown above.

But I didn't follow the step 3 `Add the Captcha frontend component`, since I don't use React.

I'm handling it on the server side.

```js
    const { data, error } = await client.auth.signInWithPassword({
      email,
      password,
      options: {
        captchaToken: 'faketoken', // for testing
      },
    })
```

Would `signInWithPassword` return an error if an invalid token was provided?

**Comment by razonyang** [comment, marked as answer]
For others ran into this, the captcha challenge will be bypassed when using a Supabase client instance with the `service_role` secret.
